### ISSUE TITLE
Race vs parallel composition

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#20]
Currently we use the Alternative `<|>` operator to compose streams in parallel. This composition runs all the streams in parallel and combines all of them in a single stream. The dual scenario is to race the streams in parallel, take the first result and cancel the rest. This is how `<|>` behaves in the async package for example. We can use two separate operators for these two behaviors. One proposal is to use `<|>` for racing and `<||>` for parallel composition, the vertical parallel bars meaning parallel, though it is a bit longer operator with four chars.

**Comment by harendra-kumar** [comment]
The race combinator can be implemented by just doing everything in parallel and use `take 1` which should cancel the pending tasks.

**Comment by harendra-kumar** [comment]
Closing in favor of #18.
